src/states/common/base.rs

```diff
@@ -452,3 +452,7 @@ pub fn to_network_bytes(message: &Message) -> Result<Bytes, RoutingError> {
 pub fn from_network_bytes(data: &Bytes) -> Result<Message, RoutingError> {
     Ok(bincode::deserialize(&data[..])?)
 }
+
+pub fn peek_from_network_bytes(data: &Bytes) -> Result<MessagePeek, RoutingError> {
+    Ok(bincode::deserialize(&data[..])?)
+}
```

Comment:
As we no longer have mock serialization, we can now change `from_network_bytes` and `to_network_bytes` to be generic. The we wouldn't need this function.

Reply:
Or maybe even remove those function altogether and use `bincode` directly.

Reply:
I'm not sure the generic option really helps.
I think it is not unreasonable to look at removing these function and use bincode directly (though I'm not sure it would make things more ergonomic/readable so I'd be incline to leave it until we decide it makes more sense?

Reply:
OK, I'm fine with postponing this one.